plugins/outputs/kafka/kafka.go

```diff
@@ -12,54 +13,107 @@ import (
 	"github.com/Shopify/sarama"
 )
 
-type Kafka struct {
-	// Kafka brokers to send metrics to
-	Brokers []string
-	// Kafka topic
-	Topic string
-	// Routing Key Tag
-	RoutingTag string `toml:"routing_tag"`
-	// Compression Codec Tag
-	CompressionCodec int
-	// RequiredAcks Tag
-	RequiredAcks int
-	// MaxRetry Tag
-	MaxRetry int
-
-	// Legacy SSL config options
-	// TLS client certificate
-	Certificate string
-	// TLS client key
-	Key string
-	// TLS certificate authority
-	CA string
-
-	// Path to CA file
-	SSLCA string `toml:"ssl_ca"`
-	// Path to host cert file
-	SSLCert string `toml:"ssl_cert"`
-	// Path to cert key file
-	SSLKey string `toml:"ssl_key"`
-
-	// Skip SSL verification
-	InsecureSkipVerify bool
-
-	// SASL Username
-	SASLUsername string `toml:"sasl_username"`
-	// SASL Password
-	SASLPassword string `toml:"sasl_password"`
-
-	tlsConfig tls.Config
-	producer  sarama.SyncProducer
-
-	serializer serializers.Serializer
+const (
+	TOPIC_SUFFIX_METHOD_EMPTY uint8 = iota
+	TOPIC_SUFFIX_METHOD_MEASUREMENT
+	TOPIC_SUFFIX_METHOD_TAG
+	TOPIC_SUFFIX_METHOD_TAGS
+)
+
+var TopicSuffixMethodStringToUID = map[string]uint8{
+	"":            TOPIC_SUFFIX_METHOD_EMPTY,
+	"measurement": TOPIC_SUFFIX_METHOD_MEASUREMENT,
+	"tag":         TOPIC_SUFFIX_METHOD_TAG,
+	"tags":        TOPIC_SUFFIX_METHOD_TAGS,
 }
```

Comment:
I would remove the map/enums, they don't really buy us anything from a type safety point of view.  You could just have a list of valid strings for validating the config file.

Reply:
You're probably right about having separator between topic and topic_suffix, I think that follows the principle of least astonishment, good catch.
Renamed key_separator to just separator also.
I'm not sure about maps/enums though, because they kinda make the code less error-prone, it's easier to make a typo in a string than in a constant name. We have testcases covering that though. (And I'm not familiar with commonly accepted practices in Go)